docs/pages/linking/into-your-app.mdx

```diff
@@ -0,0 +1,129 @@
+
+<Collapsible summary="Test a link using Expo Go">
+
+By default, [Expo Go](https://expo.dev/go) uses the `exp://` scheme. If you link to `exp://` without specifying a URL address afterward, it will open the app to the home screen. In development, your app's complete URL looks like `exp://127.0.0.1:8081`.
+
+To open the `/details` screen while testing on Expo Go, you can use `npx uri-scheme`:
+
+<Terminal
+  cmd={['$ npx uri-scheme open exp://127.0.0.1:8081/--/somepath/into/app?hello=world --ios']}
+/>
+
+In Expo Go, `/--/` is added to the URL when a path is specified. This indicates to Expo Go that the substring after it corresponds to the deep link path and is not part of the path to the app itself.
+
+By default, `exp://` is replaced with `http://` when opening a URL in Expo Go. You can also use `exps://` to open `https://` URLs. However, `exps://` does not currently support loading sites with insecure TLS certificates.
+
+</Collapsible>
+
+## Handle URLs
+
+You can observe links that launch your app using the [`Linking.useURL()`](/versions/latest/sdk/linking/#useurl) hook from [`expo-linking`](/versions/latest/sdk/linking/).
+
+```tsx index.tsx
+import * as Linking from 'expo-linking';
+
+export default function Home() {
+  const url = Linking.useURL();
+
+  return <Text>URL: {url}</Text>;
+}
+```
+
+The `Linking.useURL()` hook works behind the scenes by following these imperative methods:
+
+- The link that launched the app is initially returned using [`Linking.getInitialURL()`](/versions/latest/sdk/linking/#linkinggetinitialurl)
+- Any new links triggered while the app is already open are observed with [`Linking.addEventListener('url', callback)`](/versions/latest/sdk/linking/#linkingaddeventlistenertype-handler)
+
+## Parse URLs
+
+You can use the [`Linking.parse()`](/versions/latest/sdk/linking/#linkingparseurl) method to parse the **path**, **hostname**, and **query parameters** from a URL. This method extracts deep linking information and considers nonstandard implementations.
+
+```tsx index.tsx|collapseHeight=440
+import * as Linking from 'expo-linking';
+
+export default function Home() {
+  const url = Linking.useURL();
+
+  if (url) {
+    const { hostname, path, queryParams } = Linking.parse(url);
+
+    console.log(
+      `Linked to app with hostname: ${hostname}, path: ${path} and data: ${JSON.stringify(
+        queryParams
+      )}`
+    );
+  }
+
+  return (
+    /* @hide Your React component here. */ /* @end */
+  )
+}
+```
+
+## When to not use deep links
+
+If a user does not has your app installed and follows a link to your app's custom scheme, the device's operating system will indicate that the page couldn't be opened without providing any further information. This results in a poor user experience, and unfortunately, there's no way to work around this in a web browser.
```

Comment:
I'm not sure (after working on iOS universal links and Android App Links), the last statement is true anymore, since Apple Smart Banner handles this for iOS. I think I might not have complete context.

@brentvatne: is the last statement still relevant?

Reply:
We could walk around this a bit by being less prescriptive ... instead of "when not to use deep links", this could be "limitations" and something as simple as, "If the user doesn't have your app installed, deep links don't link to anything." and optionally "Attribution services like Branch offer solutions for conditionally linking to your app or a web page" (at least as I recall) and/ or "one solution is to use universal links, which will route to your web page when the app isn't installed [learn more]"